src/main/java/it/pagopa/pn/deliverypush/service/mapper/TimelineMapperAfterFix.java

```diff
@@ -0,0 +1,113 @@
+package it.pagopa.pn.deliverypush.service.mapper;
+
+import it.pagopa.pn.commons.exceptions.PnInternalException;
+import it.pagopa.pn.deliverypush.dto.address.LegalDigitalAddressInt;
+import it.pagopa.pn.deliverypush.dto.timeline.TimelineElementInternal;
+import it.pagopa.pn.deliverypush.dto.timeline.details.RecipientRelatedTimelineElementDetails;
+import it.pagopa.pn.deliverypush.dto.timeline.details.ScheduleRefinementDetailsInt;
+import it.pagopa.pn.deliverypush.dto.timeline.details.SendDigitalDetailsInt;
+import it.pagopa.pn.deliverypush.dto.timeline.details.TimelineElementCategoryInt;
+import it.pagopa.pn.deliverypush.exceptions.PnDeliveryPushExceptionCodes;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.util.Set;
+
+@Slf4j
+public class TimelineMapperAfterFix extends TimelineMapper {
+
+    public void remapSpecificTimelineElementData(Set<TimelineElementInternal> timelineElementInternalSet, TimelineElementInternal result, Instant ingestionTimestamp) {
+        if (result != null) {
+            //L'ingestion timestamp viene settato con il timestamp originale dell'evento (dunque timestamp evento per SEND)
+            result.setIngestionTimestamp(ingestionTimestamp);
+
+            //Nello switch case invece vengono effettuati ulteriori remapping dei timestamp, questi non dipendono dal singolo elemento, ma necessitano di tutta la timeline
+            switch (result.getCategory()) {
```

Comment:
La logica mi torna. Però approfittando delle modifiche sarebbe utile inserire un pò di commenti negli switch per le varie casistiche. I metodi chiamati già esistenti non sono parlanti, dunque sarebbe utile avere qualche commento che spieghi cosa si sta facendo e perchè (domani difficilmente ci ricorderemo di tutto il delirio analizzato in questo SRS)